### PR TITLE
Add tests for filter-empty request util

### PR DESCRIPTION
**Pull request by Baltazore**
@runspired small quick one PR here, please let me know if there is some cases I missed. Please also check if I put this file in correct place. 

there is 2 commented out fields, it fails `SerializablePrimitive` type, so I cannot even put it there... I am not sure if those should be even tested.

### CODE REVIEW

File: tests/builders/tests/unit/filter-empty-test.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { module, test } from 'qunit';
+
+import { filterEmpty } from '@ember-data/request-utils';
+
+module('filterEmpty', function () {
```
Comment (runspired) [inline tests/builders/tests/unit/filter-empty-test.ts:5]: this is probably more useful filtering only `undefined` `null` and `[]`. `false` and `0` tend to have real meaning.
`[[]]` and similar is interesting. I think we leave those, not sure how those even serialize today would have to check

File: packages/request-utils/src/index.ts [hunk 2]
```diff
@@ -350,7 +350,8 @@ export function filterEmpty(source: Record<string, Serializable>): Record<string
   const result: Record<string, Serializable> = {};
   for (const key in source) {
     const value = source[key];
-    if (value) {
+    // Allow falsy values except for undefined and null
+    if (value !== undefined && value !== null) {
```
Comment (runspired) [inline packages/request-utils/src/index.ts:354]:
```suggestion
    if (value !== undefined && value !== null && value !== '') {
```

File: packages/request-utils/src/index.ts [hunk 3]
```diff
@@ -350,7 +350,8 @@ export function filterEmpty(source: Record<string, Serializable>): Record<string
   const result: Record<string, Serializable> = {};
   for (const key in source) {
     const value = source[key];
-    if (value) {
+    // Allow falsy values except for undefined and null
```
Comment (runspired) [inline packages/request-utils/src/index.ts:353]:
```suggestion
    // Allow `0` and `false` but filter falsy values that indicate "empty"
```

File: tests/builders/tests/unit/filter-empty-test.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  test('it returns an object with truthy values and meaningful falsy values like `false` and `0`', function (assert) {
+    assert.deepEqual(
+      filterEmpty({
+        foo: 'bar',
+        baz: null,
+        zero: 0,
+        booleanFalse: false,
+        emptyArray: [],
```
Comment (runspired) [inline tests/builders/tests/unit/filter-empty-test.ts:17]:
```suggestion
        emptyString: '',
        emptyArray: [],
```
